### ISSUE TITLE
Different results in testing

### ISSUE THREAD

**Issue by ShazAlvi** [opengeokube/kit4dl#51]
**Describe the bug**
The metrics from testing (using the best model) after training and validation is different when the same best model is used loaded using the cli command `kit4dl test`

**To Reproduce**
1. Train a model, and then run the testing after the training.
2. Use the best model saved to run the test for the same dataset.
3. The results from the above two procedures would be different.  

**Expected behavior**
The metric from the test performed immediately after training and validation using the best model to have the same result as when the best model is used with the `kit4dl test` command.

**Screenshots**
Sadly no screenshots.

**Setup (please complete the following information):**
 - OS: Linux
 - Python version  Python 3.10.11
 - Conda packages (`conda list` result)
 - PIP packages  (`pip freeze` result)
 - Other relevant information

**Additional context**
None.

**Comment by ShazAlvi** [comment]
@jamesWalczak  Could you please comment on this?

**Comment by jamesWalczak** [comment]
@ShazAlvi are you sure to:

- [ ] use the same TOML configuration file?
- [ ] use the same checkpoint?

The procedures are quite the same:
1. Set-up `Trainer` using the conf TOML file
2. Prepare `Trainer`
3. Run testing
Hence, unless there is some side-effect during `fit` phase, the results should be quite the same (provided pseudo-random `seed` is set and the pytorch is in `deterministic` mode.)

**Comment by ShazAlvi** [comment]
The configuration file was the same, and also the checkpoint (I also tried the cases where there was only one checkpoint in the saved folder so there can be no confusion). I looked through the code and the procedures seemed to be the same. My only guess is that something happens when trainer.prepare() is called, but I do not have an idea what can go wrong. Otherwise, running `kit4dl test` twice gives exactly the same results.
